pkg/runner/cluster_k8s.go

```diff
@@ -249,6 +258,49 @@ func (*ClusterK8sRunner) CompatibleBuilders() []string {
 	return []string{"docker:go"}
 }
 
+func (*ClusterK8sRunner) CollectOutputs(ctx context.Context, input *api.CollectionInput, w io.Writer) error {
+	log := logging.S().With("runner", "cluster:k8s", "run_id", input.RunID)
+
+	log.Info("collecting outputs")
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(outputsBucketRegion)},
+	)
+	if err != nil {
+		return fmt.Errorf("Couldn't establish an AWS session to list items in bucket: %v", err)
+	}
+
+	svc := s3.New(sess)
+
+	resp, err := svc.ListObjectsV2(&s3.ListObjectsV2Input{Bucket: aws.String(outputsBucket), Prefix: aws.String(input.RunID)})
+	if err != nil {
+		return fmt.Errorf("Unable to list items in bucket %q, %v", outputsBucket, err)
+	}
+
+	downloader := s3manager.NewDownloader(sess)
+
+	zipWriter := zip.NewWriter(w)
+	defer zipWriter.Close()
+
+	for _, item := range resp.Contents {
+		ww, err := zipWriter.Create(*item.Key)
+		if err != nil {
+			return fmt.Errorf("Couldn't add file to the zip archive: %v", err)
+		}
+
+		_, err = downloader.Download(FakeWriterAt{ww},
```

Comment:
@raulk I wanted to add parallel downloads here, but it is hard, because the `downloader.Download` expects a `io.WriterAt`, and we are sequentially writing to the zip writer. So I am not exactly sure how we can have parallel downloads with parallel writers to the same zip archive - something to investigate if we can do better/faster.